index.d.ts

```diff
@@ -104,4 +112,5 @@ export interface ClientConfiguration {
   readonly token: string;
 }
 
-export default function createClient(config: ClientConfiguration): Client;
+export const createClient: (config: ClientConfiguration) => Client;
+export default createClient;
```

Comment:
Rather than changing the typing, I think the right way to type the existing behavior is to use 
`export = createClient`. Then adding the `createClient` function and default attributes on top of a function can be avoided.